patches/server/0677-More-Projectile-API.patch

```diff
@@ -791,3 +791,122 @@ index e8a455eb5e17bcfcae3f03664f2b47773fbdf37e..08178a88ba7d0881a6c2843eef24a846
          return ((CraftItemType<?>) CraftItemType.minecraftToBukkitNew(item.getItem())).getItemMeta(item);
      }
  
+diff --git a/src/main/java/org/bukkit/craftbukkit/projectiles/CraftBlockProjectileSource.java b/src/main/java/org/bukkit/craftbukkit/projectiles/CraftBlockProjectileSource.java
+index 8b57044518fbb0fd507da6ae3ff12b99eff70b37..b685aa613277635e46da8a47457d629bb0c1e01f 100644
+--- a/src/main/java/org/bukkit/craftbukkit/projectiles/CraftBlockProjectileSource.java
++++ b/src/main/java/org/bukkit/craftbukkit/projectiles/CraftBlockProjectileSource.java
+@@ -64,12 +64,13 @@ public class CraftBlockProjectileSource implements BlockProjectileSource {
+         Direction enumdirection = (Direction) sourceblock.state().getValue(DispenserBlock.FACING);
+         net.minecraft.world.level.Level world = this.dispenserBlock.getLevel();
+         net.minecraft.world.entity.Entity launch = null;
++        net.minecraft.world.item.Item item = null; // Paper
+ 
+         if (Snowball.class.isAssignableFrom(projectile)) {
+             launch = new net.minecraft.world.entity.projectile.Snowball(world, iposition.x(), iposition.y(), iposition.z());
+         } else if (Egg.class.isAssignableFrom(projectile)) {
+             launch = new ThrownEgg(world, iposition.x(), iposition.y(), iposition.z());
+-        } else if (EnderPearl.class.isAssignableFrom(projectile)) {
++        } else if (false && EnderPearl.class.isAssignableFrom(projectile)) { // Paper - disable enderpearl, previously would NPE and this can't be approximate
+             launch = new ThrownEnderpearl(world, null);
```

Comment:
WHy does this *have* to NPE? Can't we just use the other constructor that ends up not setting the owner?